### PR TITLE
Update for WP 6.4 compatibility, add block support

### PR DESCRIPTION
**Pull request by richienabuk**

### CODE REVIEW

File: includes/class-klp-wc-payment-gateway-blocks-support.php [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+	/**
+	 * Initializes the payment method type.
+	 */
+	public function initialize() {
+		$this->settings = get_option( 'woocommerce_klump_settings', [] );
+
+        $payment_gateways       = WC()->payment_gateways()->payment_gateways();
+        $this->gateway = $payment_gateways['klump'];
+
+//		add_action( 'woocommerce_rest_checkout_process_payment_with_context', array( $this, 'failed_payment_notice' ), 8, 2 );
```
Comment (cyberomin) [inline includes/class-klp-wc-payment-gateway-blocks-support.php]: Is this not needed again?

File: klump-wc-payment.php [hunk 2]
```diff
@@ -13,43 +13,46 @@
...
 
-if (!defined('ABSPATH')) {
+if ( ! defined('ABSPATH')) {
     exit;
 }
 
 define('KLP_WC_PLUGIN_FILE', __FILE__);
 define('KLP_WC_SDK_URL', 'https://js.useklump.com/klump.js');
-define('KLP_WC_SDK_VERIFICATION_URL', 'https://api.useklump.com/v1/transactions/');
+define('KLP_WC_SDK_VERIFICATION_URL', 'https://staging-api.useklump.com/v1/transactions/');
```
Comment (204070) [inline klump-wc-payment.php:33]: This has been updated to staging. 
```suggestion
define('KLP_WC_SDK_VERIFICATION_URL', 'https://api.useklump.com/v1/transactions/');
```

File: assets/js/blocks/checkout.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
+// const settings = window.wc.wcSettings.getSetting( 'woocommerce_klump_settings', {} );
+// const label = window.wp.htmlEntities.decodeEntities( settings.title ) || window.wp.i18n.__( 'Klump', 'wc-phonepe' );
+//
+// const Content = () => {
+//   return window.wp.htmlEntities.decodeEntities( settings.description || '' );
+// };
+//
+// const Block_Gateway = {
```
Comment (cyberomin) [inline assets/js/blocks/checkout.js]: Are these commented out code not needed again @richienabuk

File: assets/js/blocks/checkout.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+//
+// window.wc.wcBlocksRegistry.registerPaymentMethod( Block_Gateway );
+
+// Pay in 4 instalments - Klump BNPL (logo)
+
+(() => {
+  "use strict";
+  const e = window.React,
+      t = window.wc.wcBlocksRegistry,
+      l = window.wc.wcSettings,
```
Comment (cyberomin) [inline assets/js/blocks/checkout.js]: Can these variables have better names? alphabet alone doesn't help.

File: klump-wc-payment.php (hunk 2 above)
Comment (cyberomin) [inline klump-wc-payment.php:33]: Why are we showing our staging server to the world? This is supposed to be private to us and us alone.

**Review by cyberomin — APPROVED** [review]

**Pull request merged**